clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2535,6 +2535,55 @@ def GetRuntimeMemberOp : CIR_Op<"get_runtime_member"> {
   let hasVerifier = 1;
 }
 
+//===----------------------------------------------------------------------===//
+// GetMethodCalleeOp
+//===----------------------------------------------------------------------===//
+
+def GetMethodCalleeOp : CIR_Op<"get_method_callee"> {
+  let summary = "Resolve a method to a function pointer as callee";
+  let description = [{
+    The `cir.get_method_callee` operation takes a method and an object as input,
+    and yields a function pointer that points to the actual function
+    corresponding to the input method. The operation also yields an integer
+    that represents the necessary adjustment required to be applied on the
+    "this" pointer before calling the callee.
+
+    This operation is generated when calling a method through a pointer-to-
+    member-function in C++:
+
+    ```cpp
+    // Foo *object;
+    // int arg;
+    // void (Foo::*method)(int);
+
+    (object->*method)(arg);
+    ```
+
+    The code above will generate CIR similar as:
+
+    ```mlir
+    // %object = ...
+    // %arg = ...
+    // %method = ...
+    %callee, %this_adj = cir.get_method_callee %method, %object
+    %this = cir.ptr_stride %object, %this_adj
```

Comment:
Rename this to `cir.get_method`. Is there any purpose in keeping the `%this_adj` around? I wonder if we should hide the ptr_stride and return the adjusted this point instead.

Reply:
> I wonder if we should hide the ptr_stride and return the adjusted this point instead.

This is better.

Reply:
Updated.